search/searcher/search_geoshape.go

```diff
@@ -70,6 +70,20 @@ func buildRelationFilterOnShapes(ctx context.Context, dvReader index.DocValueRea
 	var dvShapeValue []byte
 	var startReading, finishReading bool
 	var reader *bytes.Reader
+
+	bufPool := make([][]byte, 11)
+	bufPool[0] = make([]byte, 8192*3)
+	bufPool[1] = make([]byte, 4096*3)
+	bufPool[2] = make([]byte, 2048*3)
+	bufPool[3] = make([]byte, 1024*3)
+	bufPool[4] = make([]byte, 512*3)
+	bufPool[5] = make([]byte, 256*3)
+	bufPool[6] = make([]byte, 128*3)
```

Comment:
As discussed, you can lazily initialise the buffer of a particular size only when there is a demand of it.